rust/src/lib.rs

```diff
@@ -79,45 +92,65 @@ use store::{
 static mut CHANGED_CALLBACK: Option<extern fn()> = None;
 
 fn transact_items_vocabulary(in_progress: &mut InProgress) -> Result<(), list_errors::Error> {
-    let schema = r#"[
-        {   :db/ident       :item/uuid
-            :db/valueType   :db.type/uuid
-            :db/cardinality :db.cardinality/one
-            :db/unique      :db.unique/value
-            :db/index true },
-        {   :db/ident       :item/name
-            :db/valueType   :db.type/string
-            :db/cardinality :db.cardinality/one
-            :db/index       true
-            :db/fulltext    true  },
-        {   :db/ident       :item/due_date
-            :db/valueType   :db.type/instant
-            :db/cardinality :db.cardinality/one  },
-        {   :db/ident       :item/completion_date
-            :db/valueType   :db.type/instant
-            :db/cardinality :db.cardinality/one  },
-        {  :db/ident     :item/label
-            :db/valueType :db.type/ref
-            :db/cardinality :db.cardinality/many }]"#;
-    in_progress.transact(schema)
+    in_progress.ensure_vocabulary(&Definition {
```

Comment:
Informal before/after timings and memory usage of this (in release mode)? I would expect a small memory saving…

Reply:
Marginally faster startup times: 
Non vocab managment
```startup time: Duration { secs: 0, nanos: 783583278 }
startup time: Duration { secs: 0, nanos: 718939582 }
startup time: Duration { secs: 0, nanos: 428831956 }
startup time: Duration { secs: 0, nanos: 61791586 }
startup time: Duration { secs: 0, nanos: 63743039 }```

Vocab management:
```startup time: Duration { secs: 0, nanos: 769641038 }
startup time: Duration { secs: 0, nanos: 679142944 }
startup time: Duration { secs: 0, nanos: 330097111 }
startup time: Duration { secs: 0, nanos: 58794550 }
startup time: Duration { secs: 0, nanos: 54428597 }```